components/google_calendar/sources/common.mjs

```diff
@@ -0,0 +1,74 @@
+import googleCalendar from "../google_calendar.app.mjs";
+
+export default {
+  props: {
+    googleCalendar,
+    calendarId: {
+      propDefinition: [
+        googleCalendar,
+        "calendarId",
+      ],
+    },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 15 * 60,
+      },
+    },
+  },
+  hooks: {
+    async activate() {
+      await this.processEvents({});
+    },
+  },
+  methods: {
+    getIntervalData(event) {
+      const intervalMs = 1000 * (event.interval_seconds || 300); // fall through to default for manual testing
```

Comment:
Just checking, if `event.interval_seconds = 0` so the evaluated value will be `300`. Maybe you want to use `??` operator or the idea is to use `300` if `event.interval_seconds = 0 or null`?